zk_toolbox/crates/zk_inception/src/commands/chain/deploy_and_bridge_zk.rs

```diff
@@ -0,0 +1,298 @@
+            "function saveL1Address() public"
+        ])
+        .unwrap(),
+    );
+}
+
+fn find_latest_json_file(directory: &str) -> Option<PathBuf> {
+    fs::read_dir(directory)
+        .ok()?
+        .flatten()
+        .map(|entry| entry.path())
+        .find(|path| {
+            path.file_name()
+                .and_then(|name| name.to_str())
+                .map_or(false, |name| name.ends_with("-latest.json"))
+        })
+}
+
+pub async fn run(args: ForgeScriptArgs, shell: &Shell) -> anyhow::Result<()> {
+    // Setup
+    let ecosystem_config = EcosystemConfig::from_file(shell)?;
+
+    let chain_name = global_config().chain_name.clone();
+    let chain_config = ecosystem_config
+        .load_chain(chain_name)
+        .context(MSG_CHAIN_NOT_INITIALIZED)?;
+
+    let l1_url = chain_config
+        .get_secrets_config()?
+        .l1
+        .context(MSG_L1_SECRETS_MUST_BE_PRESENTED)?
+        .l1_rpc_url
+        .expose_str()
+        .to_string();
+
+    let era_chain_id = U256::from(chain_config.chain_id.0);
+    let era_provider = Provider::<Http>::try_from(
+        chain_config
+            .get_general_config()
+            .unwrap()
+            .api_config
+            .unwrap()
+            .web3_json_rpc
+            .http_url,
+    )?;
+    let era_client: Client<L2> = Client::http(
+        chain_config
+            .get_general_config()
+            .unwrap()
+            .api_config
+            .unwrap()
+            .web3_json_rpc
+            .http_url
+            .parse()
+            .unwrap(),
+    )?
+    .for_network(L2::from(L2ChainId(chain_config.chain_id.0)))
+    .build();
+
+    let hash = call_script(
+        shell,
+        args.clone(),
+        &DEPLOY_AND_BRIDGE_ZK_TOKEN_INTERFACE
+            .encode(
+                "supplyEraWallet",
+                (
+                    chain_config.get_wallets_config()?.governor.address,
+                    U256::from_dec_str("10000000000000000000").unwrap(),
+                ),
+            )
+            .unwrap(),
+        &ecosystem_config,
+        chain_config.get_wallets_config()?.governor_private_key(),
+        l1_url.clone(),
+    )
+    .await?;
+
+    println!("Pausing for 20 seconds after funding wallet...");
+    tokio::time::sleep(tokio::time::Duration::from_secs(20)).await;
+    println!("Resuming execution");
+
+    // Call the `finalizeZkTokenWithdrawal` function using the extracted parameters
+    let calldata = DEPLOY_AND_BRIDGE_ZK_TOKEN_INTERFACE
+        .encode("run", ())
+        .unwrap();
+
+    let tx_hash = call_script_era(
+        shell,
+        args.clone(),
+        &calldata,
+        &ecosystem_config,
+        chain_config.get_wallets_config()?.governor_private_key(),
+        era_provider.url().to_string(),
+    )
+    .await?;
+
+    println!("ZK Token Deployed and Withdrawn to L1!");
+
+    println!("Pausing for 20 seconds...");
+    tokio::time::sleep(tokio::time::Duration::from_secs(20)).await;
```

Comment:
why do we need these 20 seconds if we below we will anyway wait for the tx to complete?